packages/hardhat-core/src/internal/core/config/config-loading.ts

```diff
@@ -73,3 +89,65 @@ Learn more about compiler configuration at https://usehardhat.com/configuration"
 
   return resolved;
 }
+
+/**
+ * Receives a parsed stack trace of a MODULE_NOT_FOUND error and checks
+ * if the error comes from a missing peer dependency from a hardhat plugin.
+ *
+ * If that's the case, it throws an error. Otherwise it does nothing.
+ */
+export function analyzeModuleNotFoundStackTrace(
+  stackTrace: stackTraceParser.StackFrame[],
```

Comment:
Maybe this should take `Error`, so that `stacktrace-parser` is only used here.

Reply:
Yes, that's a good idea, and that way it doesn't just *assume* that it's a `MODULE_NOT_FOUND`, that kind of bugged me.